artiq/device_db_testing/eems.py

```diff
@@ -0,0 +1,146 @@
+from migen import *
+from migen.build.generic_platform import *
+from migen.genlib.io import DifferentialOutput
+
+from artiq.gateware import rtio
+from artiq.gateware.rtio.phy import ttl_simple, ttl_serdes_7series, spi2
+from artiq.build_soc import build_artiq_soc
+from artiq import __version__ as artiq_version
+
+from artiq.gateware.targets.kasli import (_dio, _urukul)
+
+import json
+
+#Create PHYs and rtio channels for a dio device and return dictionary with information

```

Comment:
You will want to split this so that the definition of the RTIO channels and the `device_db` tooling is not in a single file for all EEMs but split.